src/persist-client/src/impl/compact.rs

```diff
@@ -146,13 +146,21 @@ impl Compactor {
                         return;
                     }
                 };
+                let keys = res.output.keys.clone();
```

Comment:
pretty sure we can have `Machine::merge_res` take a `&FueledMergeRes` and avoid the key copy here

Reply:
Machine doesn't have access to blob (which I sorta like)

Reply:
oh, I completely misunderstood, disregard! 🤦